public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -447,99 +450,99 @@ function getDefaultDataSourceRef(): DataSourceRef | undefined {
 }
 
 // Function to know if the dashboard transformed is a valid DashboardV2Spec
-function isDashboardSchemaV2(dash: any): dash is DashboardV2Spec {
+function validateDashboardSchemaV2(dash: any): { isValid: boolean; reason: string } {
```

Comment:
This is a validation function that we use when transforming. I got some validation errors during tests, that I found impossible to debug if we don't have the throw reason. So I added here a second parameter to have the reason. The bad part is we lose the type checking, but we could get it if we return the dashboard as DashboardV2Spec. Since we use it only in one place, I didn't do it. To not confuse anybody having a different reference returned by the validation function.

Reply:
I like the new `reason` part, very handy!